### PR TITLE
Improve /vote

### PR DESCRIPTION
**Pull request by MartinKrausewitz**
pull the updated vote function to master. Added Buttons, better code

### CODE REVIEW

**Comment by Kr0nox** [comment]
And I think we should consistent with pro/con and for/against, so only choose one, we are currently using both

**Comment by MartinKrausewitz** [comment]
voteButtonlistener is now deleted and we use for/against consistent

File: slashCommands/vote.ts [hunk 1]
```diff
@@ -12,178 +19,142 @@ export default {
...
-
-        // --- Check for undefined
-
-        time = ((time == undefined || time <= 0) ? 30 : time);
-
-        title = (title == undefined) ? "" : title;
-
-        // --- group variables
 
         const guid = get("vote_group", "config") as string;
```
Comment (c0derMo) [inline slashCommands/vote.ts:51]: "vote_group" isn't in the config-templates - and should also use camelCase

Comment (MartinKrausewitz) [reply slashCommands/vote.ts:51]: you mean voteGroup? The other config variables also follow x_x like announcement_channel

File: slashCommands/vote.ts [hunk 2]
```diff
@@ -12,178 +19,142 @@ export default {
...
-            await delay(1000);
-            if (reactionAr.length >= usedVotes && timed == false) {
-                voteAN.addFields({ name: "Enough Votes", value: "Vote ending in " + time + " Seconds" });
-                await reply.edit({embeds: [voteAN]});
-                await delay(time * 1000);
-                break;
-            }
+        if (timed) {
+            setTimeout(() => {
+                printVotes(pro, con, reply, title, collector);
```
[the remark this answers is not in the document]
Comment (MartinKrausewitz) [reply slashCommands/vote.ts:130]: this has been implemented through a discord timestamp

File: slashCommands/vote.ts (hunk 1 above)
Comment (MartinKrausewitz) [reply slashCommands/vote.ts:51]: I added it for now to the template the way it is

File: slashCommands/vote.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinKrausewitz) [reply slashCommands/vote.ts:130]: may edit the embed, when a group vote reaches the participation boundary?

**Comment by MartinKrausewitz** [comment]
All comments are resoleved except the renaming of config parameters @c0derMo @myphyk please review again.

File: slashCommands/vote.ts (hunk 1 above)
Comment (c0derMo) [reply slashCommands/vote.ts:47]: Ah, yeah, thought all our config variables are camel-cased, turns out, they arent

File: slashCommands/vote.ts [hunk 3]
```diff
@@ -12,187 +21,149 @@ export default {
...
+            if (!interaction.isButton()) {
+                return false;
+            }
+            return interaction.message.id === reply.id;
         };
 
         //reaction controller
-        const collector = reply.createReactionCollector({filter, dispose: true} as ReactionCollectorOptions);
+        const collector = reply.createMessageComponentCollector({filter});
+        console.log(typeof collector);
```
Comment (c0derMo) [inline slashCommands/vote.ts:76]: Floating console.log that should be removed

File: slashCommands/vote.ts [hunk 4]
```diff
@@ -12,187 +21,149 @@ export default {
...
-
     if (title != "") {
         msgEmbed.setDescription(title);
     }
-
     //show embed
     await reply.edit({embeds:[msgEmbed]})
+}
+
+function getEmbedOptions(title:string, msg: string, timestamp?:number):{embeds, components, fetchReply}{
```
Comment (c0derMo) [inline slashCommands/vote.ts:151]: This can just return the type `InteractionReplyOptions`

Comment (MartinKrausewitz) [reply slashCommands/vote.ts:151]: 'InteractionReplyOptions' is not assignable to parameter of type 'string | MessagePayload | MessageEditOptions' ; it also seems like MessagePayload dont support Embeds and MessageEditOptions dont support fetchReply.

File: slashCommands/vote.ts (hunk 3 above)
Comment (MartinKrausewitz) [reply slashCommands/vote.ts:76]: removed in next commit

**Review by c0derMo — APPROVED** [review]
Tested, fixed, approved

**Review by myphyk — APPROVED** [review]
@MartinKrausewitz @c0derMo 
Approved

**Pull request merged**